src/value_classes/value_id.rs

```diff
@@ -365,11 +365,16 @@ impl ValueID {
 impl fmt::Display for ValueID {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let node = Node::from_id(self.get_home_id(), self.get_node_id());
+        let mut node_name = node.get_name();
+        if node_name.is_empty() {
+            node_name = node.get_product_name();
+        }
```

Comment:
I'm not sure if I should implement something in Node instead... but I couldn't find a proper method name.